pkg/kubelet/network/dns/dns.go

```diff
@@ -230,7 +230,12 @@ func parseResolvConf(reader io.Reader) (nameservers []string, searches []string,
 			}
 		}
 		if fields[0] == "search" {
-			searches = fields[1:]
+			// Strip redundant trailing dot to avoid hitting search validation limits.
```

Comment:
This comment does not give enough info for future readers to understand the intent. Consider the amount of too-and-fro in the PR comments.
How about "Normalise search fields so the same domain with and without trailing dot will only count once, to avoid hitting search validation limits."

Reply:
Thanks for the feedback @bboreham !

Updated!